### PR TITLE
cash money changes

### PR DESCRIPTION
**Pull request by UnlegitSenpaii**
i like money!

### CODE REVIEW

**Comment by UnlegitSenpaii** [comment]
there were a few changes indeed

File: main.lua [hunk 1]
```diff
@@ -170,32 +172,71 @@ function SkipObjective()
...
+function DoStuffWithValuables()
+    local autocollet = AdvGetBool(cfgstr .. "autocollect")
+    local inflation = AdvGetBool(cfgstr .. "inflationishittinghard")
+
+    playerTransform = GetPlayerTransform(false);
+    local camera = GetCameraTransform()
+    local v = FindBodies("valuable", true)
+    for i=1,#v do
+        local body = v[i]
+        if body ~= nil then
```
Comment (SigmaSkid) [inline main.lua:189]: you don't need this, also
https://teardowngame.com/modding/api.html#IsHandleValid

File: main.lua [hunk 2]
```diff
@@ -170,32 +172,71 @@ function SkipObjective()
...
+            if GetPlayerInteractBody() ~= body then
+                SetBodyActive(body, false)
+                if not IsBodyBroken(body) then 
+                    local isValuable = HasTag(body, "valuable")
```
Comment (SigmaSkid) [inline main.lua]: why?

File: main.lua [hunk 3]
```diff
@@ -170,32 +172,71 @@ function SkipObjective()
...
+                                end
+                            end
+
+                            newTransform = camera
+                            local min, max = GetBodyBounds(body)
+                            local vecdistance = VecAdd(VecSub(max, min), Vec(0.1, 0, 0))
+                            newTransform.pos = VecAdd(camera.pos, vecdistance)
+                            SetBodyDynamic(body, false)
+                            SetBodyTransform(body, newTransform)
+                            SetBodyVelocity(body, Vec(0,0,0))
```
Comment (SigmaSkid) [inline main.lua]: you're already freezing it with setbodyactive

File: main.lua [hunk 4]
```diff
@@ -170,32 +172,71 @@ function SkipObjective()
...
---         SetString("hud.notification", "Picked up "..GetDescription(s).." worth $"..value)
---         Delete(s)
---         PlaySound(LoadSound("valuable.ogg"), GetCameraTransform().pos, 1.0, false)
--- 
---     end
--- end
+--mods aren't allowed to modify the savefile? who cares?
+
+function DoStuffWithValuables()
+    local autocollet = AdvGetBool(cfgstr .. "autocollect")
```
Comment (SigmaSkid) [inline main.lua]: ?

File: main.lua [hunk 5]
```diff
@@ -170,32 +172,71 @@ function SkipObjective()
...
+    local inflation = AdvGetBool(cfgstr .. "inflationishittinghard")
+
+    playerTransform = GetPlayerTransform(false);
```
Comment (SigmaSkid) [inline main.lua]: unused

File: main.lua [hunk 6]
```diff
@@ -170,32 +172,71 @@ function SkipObjective()
...
+                                    local shape = shapes[i]
+                                    local joints = GetShapeJoints(shape)
+                                    for i=1, #joints do
+                                        local joint = joints[i]
+                                        DetachJointFromShape(joint, shape)
+                                    end
+                                end
+                            end
+
+                            newTransform = camera
```
Comment (SigmaSkid) [inline main.lua]: arrays in lua are pointers

File: main.lua [hunk 7]
```diff
@@ -170,32 +172,71 @@ function SkipObjective()
...
+                            local min, max = GetBodyBounds(body)
+                            local vecdistance = VecAdd(VecSub(max, min), Vec(0.1, 0, 0))
+                            newTransform.pos = VecAdd(camera.pos, vecdistance)
```
Comment (SigmaSkid) [inline main.lua]: this modifies the local camera transform

File: main.lua (hunk 1 above)
Comment (UnlegitSenpaii) [reply main.lua:189]: i pasted it from the game. if the game is using it, im using it

File: main.lua (hunk 2 above)
Comment (UnlegitSenpaii) [reply main.lua]: i pasted it from the game. if the game is using it, im using it

File: main.lua (hunk 3 above)
Comment (UnlegitSenpaii) [reply main.lua]: no i want it static so they line up

Comment (UnlegitSenpaii) [reply main.lua]: no idea if thats causing it tho

File: main.lua (hunk 4 above)
Comment (UnlegitSenpaii) [reply main.lua]: i edited your comment

File: main.lua (hunk 7 above)
Comment (UnlegitSenpaii) [reply main.lua]: no

**Comment by UnlegitSenpaii** [comment]
gaming
merge pls

**Comment by UnlegitSenpaii** [comment]
we gaming now

**Comment by UnlegitSenpaii** [comment]
i want to play battlefield please

**Comment by SigmaSkid** [comment]
Add your features to readme and workshop description

**Review by UnlegitSenpaii — COMMENTED** [review]
i will have to approve this

**Comment by SigmaSkid** [comment]
TODO before steam workshop update:
Fix the teleport valuables positions. 
After disabling "inflation" restore valuables original value.

**Pull request merged**